web-client/src/ustc-ui/Button/Button.tsx

```diff
@@ -44,8 +49,31 @@ export const Button = props => {
     iconColor && `fa-icon-${iconColor}`,
   );
 
+  const debouncedWrapper = debounce(async () => {
+    const results = onClick();
+    if (!(results instanceof Promise))
+      throw new Error('Convert onClick method to async');
+
+    await results.finally(() => {
+      setDisableButton(false);
+    });
+  }, 500);
```

Comment:
Sorry to comment on a draft PR--I just saw the update and was excited about having a debounced button component, then I noticed my previous comment might have misguided you slightly! I'm guessing you took the 500 ms from the PR I linked in my comment, but there is certainly nothing sacred about 500 ms, and it might even make certain actions feel a little sluggish. (In my use case, a 500 ms hit wasn't a big deal because it was a one-time account creation button.)

Reply:
500 ms should be enough for the UI to disable the button;